tfe/resource_tfe_workspace.go

```diff
@@ -671,7 +676,29 @@ func resourceTFEWorkspaceDelete(d *schema.ResourceData, meta interface{}) error
 	id := d.Id()
 
 	log.Printf("[DEBUG] Delete workspace %s", id)
-	err := tfeClient.Workspaces.DeleteByID(ctx, id)
+
+	ws, err := tfeClient.Workspaces.ReadByID(ctx, id)
+	if err != nil {
+		if err == tfe.ErrResourceNotFound {
+			return nil
+		}
+		return fmt.Errorf(
+			"Error reading workspace %s: %w", id, err)
+	}
+
+	forceDelete := d.Get("force_delete").(bool)
+
+	if ws.Permissions.CanForceDelete == nil && !forceDelete {
```

Comment:
This permission could also be false. Am I right that if the two disagree then the API will handle the error and return a detailed message that after the attempt is made?

Reply:
this permission can def be false! we're relying on the atlas side to reject the force deletes because we can provide better error messaging and stay consistent with handling permissions on the atlas side as well!